internal/codecs/array.go

```diff
@@ -42,20 +42,31 @@ func popArrayCodec(
 
 // Array is an EdgeDB array type codec.
 type Array struct {
-	id    types.UUID
-	child Codec
-	typ   reflect.Type
-	step  int
+	id         types.UUID
+	child      Codec
+	typ        reflect.Type
+	step       int
+	useReflect bool
```

Comment:
Can you add some comments? It's unclear what `step` is or why `useReflect` is necessary

Reply:
added :heavy_check_mark: